libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -542,7 +555,7 @@ proc subscribe*(p: PubSub, topic: string, handler: TopicHandler) {.public.} =
 
 method publish*(
     p: PubSub, topic: string, data: seq[byte]
-): Future[int] {.base, async, public.} =
+): Future[int] {.base, async: (raises: [LPError]), public.} =
```

Comment:
a function _like_ publish should probably not be raising - ie it's a "fire-and-forget" kind of function that attempts to publish things to as many peers as possible but just like udp, there are no guarantees that it will reach anybody - as such, it doesn't _really_ make sense that it raises, *I think*.

Reply:
`publish` can raise a `LPError` if `anonimize` is `false` and we enter this branch of code:
https://github.com/vacp2p/nim-libp2p/blob/c5aa3736f96e4d66f6aa653a2351ded74b7d21a9/libp2p/protocols/pubsub/rpc/message.nim#L85-L86

This can happen if passing a nil `peerInfo` when calling `newSwitch`, (although if `SwitchBuilder` is used, it will avoid that scenario, since `build` will populate the `peerInfo` anyway)